diff --git a/sqlparse/filters/others.py b/sqlparse/filters/others.py
index da7c0e7..3388a78 100644
--- a/sqlparse/filters/others.py
+++ b/sqlparse/filters/others.py
@@ -91,6 +91,10 @@ class StripWhitespaceFilter:
             tlist.tokens.pop(1)
         while tlist.tokens[-2].is_whitespace:
             tlist.tokens.pop(-2)
+        if tlist.tokens[-2].is_group:
+            # save to remove the last whitespace
+            while tlist.tokens[-2].tokens[-1].is_whitespace:
+                tlist.tokens[-2].tokens.pop(-1)
         self._stripws_default(tlist)
 
     def process(self, stmt, depth=0):
