diff --git a/sqlparse/engine/grouping.py b/sqlparse/engine/grouping.py
index c486318..9190797 100644
--- a/sqlparse/engine/grouping.py
+++ b/sqlparse/engine/grouping.py
@@ -139,7 +139,12 @@ def group_typed_literal(tlist):
 
 def group_period(tlist):
     def match(token):
-        return token.match(T.Punctuation, '.')
+        for ttype, value in ((T.Punctuation, '.'),
+                             (T.Operator, '->'),
+                             (T.Operator, '->>')):
+            if token.match(ttype, value):
+                return True
+        return False
 
     def valid_prev(token):
         sqlcls = sql.SquareBrackets, sql.Identifier
@@ -153,7 +158,7 @@ def group_period(tlist):
     def post(tlist, pidx, tidx, nidx):
         # next_ validation is being performed here. issue261
         sqlcls = sql.SquareBrackets, sql.Function
-        ttypes = T.Name, T.String.Symbol, T.Wildcard
+        ttypes = T.Name, T.String.Symbol, T.Wildcard, T.String.Single
         next_ = tlist[nidx] if nidx is not None else None
         valid_next = imt(next_, i=sqlcls, t=ttypes)
 
