diff --git a/sqlparse/keywords.py b/sqlparse/keywords.py
index 29f27af..0f3a459 100644
--- a/sqlparse/keywords.py
+++ b/sqlparse/keywords.py
@@ -16,6 +16,7 @@ def is_keyword(value):
             or KEYWORDS_ORACLE.get(val)
             or KEYWORDS_PLPGSQL.get(val)
             or KEYWORDS_HQL.get(val)
+            or KEYWORDS_MSACCESS.get(val)
             or KEYWORDS.get(val, tokens.Name)), value
 
 
@@ -962,3 +963,8 @@ KEYWORDS_HQL = {
     'BREAK': tokens.Keyword,
     'LEAVE': tokens.Keyword,
 }
+
+
+KEYWORDS_MSACCESS = {
+    'DISTINCTROW': tokens.Keyword,
+}
