diff --git a/sqlparse/engine/statement_splitter.py b/sqlparse/engine/statement_splitter.py
index 5b3a0d9..6c69d30 100644
--- a/sqlparse/engine/statement_splitter.py
+++ b/sqlparse/engine/statement_splitter.py
@@ -17,6 +17,7 @@ class StatementSplitter:
     def _reset(self):
         """Set the filter attributes to its default values"""
         self._in_declare = False
+        self._in_case = False
         self._is_create = False
         self._begin_depth = 0
 
@@ -58,16 +59,18 @@ class StatementSplitter:
                 return 1
             return 0
 
-        # Should this respect a preceding BEGIN?
-        # In CASE ... WHEN ... END this results in a split level -1.
-        # Would having multiple CASE WHEN END and a Assignment Operator
-        # cause the statement to cut off prematurely?
+        # BEGIN and CASE/WHEN both end with END
         if unified == 'END':
-            self._begin_depth = max(0, self._begin_depth - 1)
+            if not self._in_case:
+                self._begin_depth = max(0, self._begin_depth - 1)
+            else:
+                self._in_case = False
             return -1
 
         if (unified in ('IF', 'FOR', 'WHILE', 'CASE')
                 and self._is_create and self._begin_depth > 0):
+            if unified == 'CASE':
+                self._in_case = True
             return 1
 
         if unified in ('END IF', 'END FOR', 'END WHILE'):
