diff --git a/sqlparse/cli.py b/sqlparse/cli.py
index 7a8aacb..a2ee493 100755
--- a/sqlparse/cli.py
+++ b/sqlparse/cli.py
@@ -138,7 +138,14 @@ def create_parser():
         default=False,
         type=bool,
         help='Insert linebreak before comma (default False)')
-
+    
+    group.add_argument(
+        '--compact',
+        dest='compact',
+        default=False,
+        type=bool,
+        help='Try to produce more compact output (default False)')
+    
     group.add_argument(
         '--encoding',
         dest='encoding',
diff --git a/sqlparse/filters/reindent.py b/sqlparse/filters/reindent.py
index 9fb232f..ee4c72c 100644
--- a/sqlparse/filters/reindent.py
+++ b/sqlparse/filters/reindent.py
@@ -12,7 +12,7 @@ from sqlparse.utils import offset, indent
 class ReindentFilter:
     def __init__(self, width=2, char=' ', wrap_after=0, n='\n',
                  comma_first=False, indent_after_first=False,
-                 indent_columns=False):
+                 indent_columns=False, compact=False):
         self.n = n
         self.width = width
         self.char = char
@@ -21,6 +21,7 @@ class ReindentFilter:
         self.wrap_after = wrap_after
         self.comma_first = comma_first
         self.indent_columns = indent_columns
+        self.compact = compact
         self._curr_stmt = None
         self._last_stmt = None
         self._last_func = None
@@ -196,15 +197,20 @@ class ReindentFilter:
         with offset(self, self._get_offset(tlist[0])):
             with offset(self, self._get_offset(first)):
                 for cond, value in iterable:
-                    token = value[0] if cond is None else cond[0]
-                    tlist.insert_before(token, self.nl())
+                    str_cond = ''.join(str(x) for x in cond or [])
+                    str_value = ''.join(str(x) for x in value)
+                    end_pos = self.offset + 1 + len(str_cond) + len(str_value)
+                    if (not self.compact 
+                        and end_pos > self.wrap_after):
+                        token = value[0] if cond is None else cond[0]
+                        tlist.insert_before(token, self.nl())
 
                 # Line breaks on group level are done. let's add an offset of
                 # len "when ", "then ", "else "
                 with offset(self, len("WHEN ")):
                     self._process_default(tlist)
             end_idx, end = tlist.token_next_by(m=sql.Case.M_CLOSE)
-            if end_idx is not None:
+            if end_idx is not None and not self.compact:
                 tlist.insert_before(end_idx, self.nl())
 
     def _process_values(self, tlist):
diff --git a/sqlparse/formatter.py b/sqlparse/formatter.py
index 1d1871c..761a885 100644
--- a/sqlparse/formatter.py
+++ b/sqlparse/formatter.py
@@ -115,6 +115,11 @@ def validate_options(options):
     if comma_first not in [True, False]:
         raise SQLParseError('comma_first requires a boolean value')
     options['comma_first'] = comma_first
+    
+    compact = options.get('compact', False)
+    if compact not in [True, False]:
+        raise SQLParseError('compact requires a boolean value')
+    options['compact'] = compact
 
     right_margin = options.get('right_margin')
     if right_margin is not None:
@@ -171,7 +176,8 @@ def build_filter_stack(stack, options):
                 indent_after_first=options['indent_after_first'],
                 indent_columns=options['indent_columns'],
                 wrap_after=options['wrap_after'],
-                comma_first=options['comma_first']))
+                comma_first=options['comma_first'],
+                compact=options['compact'],))
 
     if options.get('reindent_aligned', False):
         stack.enable_grouping()
