diff --git a/sqlparse/keywords.py b/sqlparse/keywords.py
index 82e39ad..d734bdf 100644
--- a/sqlparse/keywords.py
+++ b/sqlparse/keywords.py
@@ -288,7 +288,6 @@ KEYWORDS = {
     'GLOBAL': tokens.Keyword,
     'GO': tokens.Keyword,
     'GOTO': tokens.Keyword,
-    'GRANT': tokens.Keyword,
     'GRANTED': tokens.Keyword,
     'GROUPING': tokens.Keyword,
 
@@ -477,7 +476,6 @@ KEYWORDS = {
     'RETURNED_SQLSTATE': tokens.Keyword,
     'RETURNING': tokens.Keyword,
     'RETURNS': tokens.Keyword,
-    'REVOKE': tokens.Keyword,
     'RIGHT': tokens.Keyword,
     'ROLE': tokens.Keyword,
     'ROLLBACK': tokens.Keyword.DML,
@@ -577,7 +575,6 @@ KEYWORDS = {
     'TRIGGER_SCHEMA': tokens.Keyword,
     'TRIM': tokens.Keyword,
     'TRUE': tokens.Keyword,
-    'TRUNCATE': tokens.Keyword,
     'TRUSTED': tokens.Keyword,
     'TYPE': tokens.Keyword,
 
@@ -684,6 +681,9 @@ KEYWORDS_COMMON = {
     'DROP': tokens.Keyword.DDL,
     'CREATE': tokens.Keyword.DDL,
     'ALTER': tokens.Keyword.DDL,
+    'TRUNCATE': tokens.Keyword.DDL,
+    'GRANT': tokens.Keyword.DCL,
+    'REVOKE': tokens.Keyword.DCL,
 
     'WHERE': tokens.Keyword,
     'FROM': tokens.Keyword,
