diff --git a/tomlkit/__init__.py b/tomlkit/__init__.py
index 1236d01..1f2bd02 100644
--- a/tomlkit/__init__.py
+++ b/tomlkit/__init__.py
@@ -27,7 +27,7 @@ from tomlkit.api import value
 from tomlkit.api import ws
 
 
-__version__ = "0.12.0"
+__version__ = "0.12.1"
 __all__ = [
     "aot",
     "array",
diff --git a/tomlkit/items.py b/tomlkit/items.py
index 853754a..75e8ae2 100644
--- a/tomlkit/items.py
+++ b/tomlkit/items.py
@@ -612,6 +612,9 @@ class Integer(Item, _CustomInt):
 
     __int__ = unwrap
 
+    def __hash__(self) -> int:
+        return hash(self.unwrap())
+
     @property
     def discriminant(self) -> int:
         return 2
@@ -693,6 +696,9 @@ class Float(Item, _CustomFloat):
 
     __float__ = unwrap
 
+    def __hash__(self) -> int:
+        return hash(self.unwrap())
+
     @property
     def discriminant(self) -> int:
         return 3
