diff --git a/tomlkit/items.py b/tomlkit/items.py
index c2ec6e4..b80e9f1 100644
--- a/tomlkit/items.py
+++ b/tomlkit/items.py
@@ -1313,10 +1313,14 @@ class Array(Item, _CustomList):
     def __len__(self) -> int:
         return list.__len__(self)
 
+    def item(self, index: int) -> Item:
+        rv = list.__getitem__(self, index)
+        return cast(Item, rv)
+
     def __getitem__(self, key: int | slice) -> Any:
-        rv = cast(Item, list.__getitem__(self, key))
-        if rv.is_boolean():
-            return bool(rv)
+        rv = list.__getitem__(self, key)
+        if isinstance(rv, Bool):
+            return rv.value
         return rv
 
     def __setitem__(self, key: int | slice, value: Any) -> Any:
@@ -1479,6 +1483,9 @@ class AbstractTable(Item, _CustomDict):
 
         return self
 
+    def item(self, key: Key | str) -> Item:
+        return self._value.item(key)
+
     def setdefault(self, key: Key | str, default: Any) -> Any:
         super().setdefault(key, default)
         return self[key]
