diff --git a/tomlkit/container.py b/tomlkit/container.py
index 040bde7..0494019 100644
--- a/tomlkit/container.py
+++ b/tomlkit/container.py
@@ -1218,9 +1218,12 @@ def ends_with_whitespace(it: Any) -> bool:
     """Returns ``True`` if the given item ``it`` is a ``Table`` or ``AoT`` object
     ending with a ``Whitespace``.
     """
-    return (
-        isinstance(it, Table) and isinstance(it.value._previous_item(), Whitespace)
-    ) or (isinstance(it, AoT) and len(it) > 0 and isinstance(it[-1], Whitespace))
+    if isinstance(it, Whitespace):
+        return True
+    if isinstance(it, Table):
+        previous = it.value._previous_item()
+        return previous is not None and ends_with_whitespace(previous)
+    return isinstance(it, AoT) and len(it) > 0 and ends_with_whitespace(it[-1])
 
 
 def _equal_with_nan(left: Any, right: Any) -> bool:
