diff --git a/tomlkit/container.py b/tomlkit/container.py
index 35f8aec..9fb449b 100644
--- a/tomlkit/container.py
+++ b/tomlkit/container.py
@@ -521,12 +521,19 @@ class Container(_CustomDict):
             if prefix is not None:
                 _key = prefix + "." + _key
 
-        if not table.is_super_table() or (
-            any(
-                not isinstance(v, (Table, AoT, Whitespace, Null))
-                for _, v in table.value.body
+        if (
+            not table.is_super_table()
+            or (
+                any(
+                    not isinstance(v, (Table, AoT, Whitespace, Null))
+                    for _, v in table.value.body
+                )
+                and not key.is_dotted()
+            )
+            or (
+                any(k.is_dotted() for k, v in table.value.body if isinstance(v, Table))
+                and not key.is_dotted()
             )
-            and not key.is_dotted()
         ):
             open_, close = "[", "]"
             if table.is_aot_element():
