diff --git a/tomlkit/items.py b/tomlkit/items.py
index c483eed..4caf559 100644
--- a/tomlkit/items.py
+++ b/tomlkit/items.py
@@ -714,9 +714,8 @@ class Float(Item, _CustomFloat):
     def _new(self, result):
         raw = str(result)
 
-        if self._sign:
-            sign = "+" if result >= 0 else "-"
-            raw = sign + raw
+        if self._sign and result >= 0:
+            raw = f"+{raw}"
 
         return Float(result, self._trivia, raw)
 
