diff --git a/tomlkit/container.py b/tomlkit/container.py
index ecac3f1..06e6483 100644
--- a/tomlkit/container.py
+++ b/tomlkit/container.py
@@ -794,6 +794,8 @@ class OutOfOrderTableProxy(_CustomDict):
         self._tables = []
         self._tables_map = {}
 
+        original_parsing = container._parsed
+        container.parsing(True)
         for i in indices:
             _, item = self._container._body[i]
 
@@ -805,7 +807,7 @@ class OutOfOrderTableProxy(_CustomDict):
                     self._tables_map[k] = table_idx
                     if k is not None:
                         dict.__setitem__(self, k.key, v)
-
+        container.parsing(original_parsing)
         self._internal_container._validate_out_of_order_table()
 
     def unwrap(self) -> str:
