diff --git a/tomlkit/check.py b/tomlkit/check.py
new file mode 100644
index 0000000..6d9327a
--- /dev/null
+++ b/tomlkit/check.py
@@ -0,0 +1,12 @@
+def is_tomlkit(v):
+    from .container import Container
+    from .container import OutOfOrderTableProxy
+    from .items import Item as _Item
+
+    if isinstance(v, _Item):
+        return True
+    if isinstance(v, Container):
+        return True
+    if isinstance(v, OutOfOrderTableProxy):
+        return True
+    return False
diff --git a/tomlkit/container.py b/tomlkit/container.py
index 2db19e3..5b03d5e 100644
--- a/tomlkit/container.py
+++ b/tomlkit/container.py
@@ -10,6 +10,7 @@ from typing import Union
 
 from ._compat import decode
 from ._utils import merge_dicts
+from .check import is_tomlkit
 from .exceptions import KeyAlreadyPresent
 from .exceptions import NonExistentKey
 from .exceptions import TOMLKitError
@@ -46,6 +47,25 @@ class Container(_CustomDict):
     def body(self) -> List[Tuple[Optional[Key], Item]]:
         return self._body
 
+    def unwrap(self) -> str:
+        unwrapped = {}
+        for k, v in self.items():
+            if k is None:
+                continue
+
+            if not isinstance(k, str):
+                k = k.key
+
+            if isinstance(v, Item):
+                v = v.unwrap()
+
+            if k in unwrapped:
+                merge_dicts(unwrapped[k], v)
+            else:
+                unwrapped[k] = v
+
+        return unwrapped
+
     @property
     def value(self) -> Dict[Any, Any]:
         d = {}
@@ -796,6 +816,9 @@ class OutOfOrderTableProxy(_CustomDict):
                     if k is not None:
                         dict.__setitem__(self, k.key, v)
 
+    def unwrap(self) -> str:
+        return self._internal_container.unwrap()
+
     @property
     def value(self):
         return self._internal_container.value
diff --git a/tomlkit/items.py b/tomlkit/items.py
index bdc97f2..7bf88f6 100644
--- a/tomlkit/items.py
+++ b/tomlkit/items.py
@@ -27,6 +27,7 @@ from ._compat import PY38
 from ._compat import decode
 from ._utils import CONTROL_CHARS
 from ._utils import escape_string
+from .check import is_tomlkit
 from .exceptions import InvalidStringError
 from .toml_char import TOMLChar
 
@@ -492,6 +493,10 @@ class Item:
         """The TOML representation"""
         raise NotImplementedError()
 
+    def unwrap(self):
+        """Returns as pure python object (ppo)"""
+        raise NotImplementedError()
+
     # Helpers
 
     def comment(self, comment: str) -> "Item":
@@ -610,6 +615,9 @@ class Integer(int, Item):
         if re.match(r"^[+\-]\d+$", raw):
             self._sign = True
 
+    def unwrap(self) -> int:
+        return int(self)
+
     @property
     def discriminant(self) -> int:
         return 2
@@ -678,6 +686,9 @@ class Float(float, Item):
         if re.match(r"^[+\-].+$", raw):
             self._sign = True
 
+    def unwrap(self) -> float:
+        return float(self)
+
     @property
     def discriminant(self) -> int:
         return 3
@@ -739,6 +750,9 @@ class Bool(Item):
 
         self._value = bool(t)
 
+    def unwrap(self) -> bool:
+        return bool(self)
+
     @property
     def discriminant(self) -> int:
         return 4
@@ -821,6 +835,21 @@ class DateTime(Item, datetime):
 
         self._raw = raw or self.isoformat()
 
+    def unwrap(self) -> datetime:
+        (
+            year,
+            month,
+            day,
+            hour,
+            minute,
+            second,
+            microsecond,
+            tzinfo,
+            _,
+            _,
+        ) = self._getstate()
+        return datetime(year, month, day, hour, minute, second, microsecond, tzinfo)
+
     @property
     def discriminant(self) -> int:
         return 5
@@ -924,6 +953,10 @@ class Date(Item, date):
 
         self._raw = raw
 
+    def unwrap(self) -> date:
+        (year, month, day, _, _) = self._getstate()
+        return date(year, month, day)
+
     @property
     def discriminant(self) -> int:
         return 6
@@ -996,6 +1029,10 @@ class Time(Item, time):
 
         self._raw = raw
 
+    def unwrap(self) -> datetime:
+        (hour, minute, second, microsecond, tzinfo, _, _) = self._getstate()
+        return time(hour, minute, second, microsecond, tzinfo)
+
     @property
     def discriminant(self) -> int:
         return 7
@@ -1051,6 +1088,15 @@ class Array(Item, _CustomList):
         self._multiline = multiline
         self._reindex()
 
+    def unwrap(self) -> str:
+        unwrapped = []
+        for v in self:
+            if is_tomlkit(v):
+                unwrapped.append(v.unwrap())
+            else:
+                unwrapped.append(v)
+        return unwrapped
+
     @property
     def discriminant(self) -> int:
         return 8
@@ -1295,6 +1341,21 @@ class AbstractTable(Item, _CustomDict):
             if k is not None:
                 dict.__setitem__(self, k.key, v)
 
+    def unwrap(self):
+        unwrapped = {}
+        for k in self:
+            if is_tomlkit(k):
+                nk = k.unwrap()
+            else:
+                nk = k
+            if is_tomlkit(self[k]):
+                nv = self[k].unwrap()
+            else:
+                nv = self[k]
+            unwrapped[nk] = nv
+
+        return unwrapped
+
     @property
     def value(self) -> "container.Container":
         return self._value
@@ -1617,6 +1678,9 @@ class String(str, Item):
         self._t = t
         self._original = original
 
+    def unwrap(self) -> str:
+        return self.as_string()
+
     @property
     def discriminant(self) -> int:
         return 11
@@ -1675,6 +1739,15 @@ class AoT(Item, _CustomList):
         for table in body:
             self.append(table)
 
+    def unwrap(self) -> str:
+        unwrapped = []
+        for t in self._body:
+            if isinstance(t, Item):
+                unwrapped.append(t.unwrap())
+            else:
+                unwrapped.append(t)
+        return unwrapped
+
     @property
     def body(self) -> List[Table]:
         return self._body
@@ -1766,6 +1839,9 @@ class Null(Item):
     def __init__(self) -> None:
         pass
 
+    def unwrap(self) -> str:
+        return None
+
     @property
     def discriminant(self) -> int:
         return -1
